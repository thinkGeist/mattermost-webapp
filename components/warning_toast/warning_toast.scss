.warning-toast {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 10px 11px 10px 11px;
    border: 1px solid rgba(var(--center-channel-color-rgb), 0.16);
    background-color: var(--center-channel-text);
    border-radius: 4px;
    box-shadow: 0 6px 14px rgba(0, 0, 0, 0.12);
    color: var(--sidebar-text);
    font-size: 14px;
    font-weight: bold;

    .warning-toast__message {
        display: flex;
        flex-direction: row;
        line-height: 20px;
        align-items: center;

        svg {
            margin-right: 7px;
            fill: var(--sidebar-text);
        }
    }

    .warning-toast__dismiss {
        display: inline-block;
        margin-left: 30px;
        cursor: pointer;

        &:hover {
            // background-color: var(--center-channel-bg);

            svg {
                fill: var(--sidebar-text);
            }
        }

        svg {
            fill: rgba(var(--sidebar-text-rgb), 0.56);
            vertical-align: middle;
        }
    }
}

.warning-toast__wrapper {
    position: absolute;
    z-index: 3;
    display: flex;
    width: 100%;
    align-items: center;
    justify-content: center;
    bottom: 16px;
}
