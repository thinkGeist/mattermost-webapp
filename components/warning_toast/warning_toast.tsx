// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';

import {CloseIcon, AlertOutlineIcon} from '@mattermost/compass-icons/components';

import './warning_toast.scss';

type Props = {
    children: React.ReactNode;
    dismissAfterSeconds?: number;
    onDismiss: () => void;
}

export const WarningToast: React.FC<Props> = ({children, onDismiss, dismissAfterSeconds = 5}: Props) => {
    const handleDismiss = () => {
        if (typeof onDismiss === 'function') {
            onDismiss();
        }
    };

    setTimeout(() => {
        handleDismiss();
    }, 1000 * dismissAfterSeconds);

    return (
        <div className='warning-toast__wrapper'>
            <div className='warning-toast'>
                <div
                    className='warning-toast__message'
                >
                    <AlertOutlineIcon
                        size={18}
                    />
                    {children}
                </div>
                <div
                    className='warning-toast__dismiss'
                    onClick={handleDismiss}
                    data-testid='dismissWarningToast'
                >
                    <CloseIcon
                        size={18}
                    />
                </div>
            </div>
        </div>
    );
};
