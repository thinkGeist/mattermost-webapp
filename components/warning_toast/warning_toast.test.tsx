// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import {WarningToast} from './warning_toast';

describe('components/WarningToast', () => {
    test('should match snapshot', () => {
        const wrapper = shallow(
            <WarningToast
                onDismiss={jest.fn()}
            >{'A hint'}</WarningToast>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should fire onDismiss callback', () => {
        const dismissHandler = jest.fn();
        const wrapper = shallow(
            <WarningToast
                onDismiss={dismissHandler}
            >{'A hint'}</WarningToast>,
        );

        wrapper.find('.warning-toast__dismiss').simulate('click');

        expect(dismissHandler).toHaveBeenCalled();
    });

    test('should automatically dismiss after 5 seconds', () => {
        jest.useFakeTimers();
        const dismissHandler = jest.fn();
        shallow(
            <WarningToast
                onDismiss={dismissHandler}
            >{'A hint'}
            </WarningToast>,
        );

        jest.advanceTimersByTime(6000);

        expect(dismissHandler).toHaveBeenCalled();
    });

    test('should automatically dismiss after X seconds', () => {
        jest.useFakeTimers();
        const dismissHandler = jest.fn();
        shallow(
            <WarningToast
                onDismiss={dismissHandler}
                dismissAfterSeconds={10}
            >{'A hint'}
            </WarningToast>,
        );

        jest.advanceTimersByTime(11000);

        expect(dismissHandler).toHaveBeenCalled();
    });
});
