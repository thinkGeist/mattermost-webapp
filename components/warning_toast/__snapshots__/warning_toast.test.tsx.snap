// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/WarningToast should match snapshot 1`] = `
<div
  className="warning-toast__wrapper"
>
  <div
    className="warning-toast"
  >
    <div
      className="warning-toast__message"
    >
      <AlertOutlineIcon
        size={18}
      />
      A hint
    </div>
    <div
      className="warning-toast__dismiss"
      data-testid="dismissWarningToast"
      onClick={[Function]}
    >
      <CloseIcon
        size={18}
      />
    </div>
  </div>
</div>
`;
